esrally/resources/track-schema.json

```diff
@@ -207,6 +207,61 @@
       "type": "object",
       "description": "Meta-information which will be added to each metrics-record of this track."
     },
+    "component_templates": {
```

Comment:
By convention we use hyphens for user-facing names if possible (sometimes it's not). Can we rename this to `component-templates`? Also, the loader code actually uses the hyphenated name already.